File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/cart/dto/CartAddRequest.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CartAddRequest(
        @NotBlank(message = "product id is required")
        String productId,

        @NotNull(message = "quantity is required")
        @Min(value = 1, message = "quantity must be at least 1")
        Integer quantity) {
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/cart/dto/CartAddRequest.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.dto

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank

data class CartAddRequest(
    @field:NotBlank(message = "product id is required")
    var productId: String,

    @field:Min(value = 1, message = "quantity must be at least 1")
    val quantity: Int
)